src/util/dom.js

```diff
@@ -53,19 +53,19 @@ exports.contains = contains
 
 let accumulatedCSS = "", cssNode = null
 
-function insertCSS(css) {
-  if (cssNode) cssNode.textContent += css
+function insertCSS(pm, css) {
+  if (pm.cssNode) pm.cssNode.textContent += css
   else accumulatedCSS += css
 }
 exports.insertCSS = insertCSS
 
 // This is called when a ProseMirror instance is created, to ensure
 // the CSS is in the DOM.
-function ensureCSSAdded() {
-  if (!cssNode) {
-    cssNode = document.createElement("style")
-    cssNode.textContent = "/* ProseMirror CSS */\n" + accumulatedCSS
-    document.head.insertBefore(cssNode, document.head.firstChild)
+function ensureCSSAdded(pm) {
+  if (!pm.cssNode) {
```

Comment:
I think we want this logic to be a bit different. We should only add the `<style>` to the DOM once per scope. So I would create a style element with a specific class name or ID (not sure if multiple elements with same ID is allowed in the spec, if there's only one per scope — I would imagine it's permitted). That way you can determine if the CSS has been added by querying `pm.root` for a style element with the correct name.

The approach you've done will blindly add the style element once per prose mirror instance.